src/covid_model_seiir_pipeline/ode_fit/workflow.py

```diff
@@ -66,9 +58,9 @@ def __init__(self, fit_specification: FitSpecification):
             resume=True
         )
 
-    def attach_ode_tasks(self):
-        for draw_id in range(self.fit_specification.parameters.n_draws):
-            task = self._ode_fit_task_template.get_task(draw_id, self.fit_specification.data.output_root)
+    def attach_ode_tasks(self, n_draws: int):
```

Comment:
on the one hand, i strongly prefer explicit

on the other hand, now it seems there's redundant data being passed in since it's still available from the fit spec

how do you decide when you to use the more error-prone implicit internal data vs using explicit parameters that are, arguably, redundant?

Reply:
The workflow no long hangs on to the fit specification.  It only gets the info it needs (a version) on init.

Reply:
So no redundancy.  I'm making the specifications abstractions for use by cli/main/task code only as a way to communicate parameters.  Executor code and modeling code will get primative types whereever possible (or potentially a data container where necessary).